Description: Fix UnicodeError in sync_members
Origin: https://bazaar.launchpad.net/~mailman-coders/mailman/2.1/revision/1429
 and https://bazaar.launchpad.net/~mailman-coders/mailman/2.1/revision/1437
Bug-Debian: #732741
Forwarded: not-needed
Applied-Upstream: in 2.1.18 (2.1.17 contains the bug!)

--- a/bin/sync_members
+++ b/bin/sync_members
@@ -1,6 +1,6 @@
 #! @PYTHON@
 #
-# Copyright (C) 1998-2003 by the Free Software Foundation, Inc.
+# Copyright (C) 1998-2013 by the Free Software Foundation, Inc.
 #
 # This program is free software; you can redistribute it and/or
 # modify it under the terms of the GNU General Public License
@@ -256,6 +256,10 @@ def main():
             try:
                 if not dryrun:
                     mlist.ApprovedAddMember(userdesc, welcome, notifyadmin)
+                # Avoid UnicodeError if name can't be decoded
+                if isinstance(name, str):
+                    name = unicode(name, errors='replace')
+                name = name.encode(enc, 'replace')
                 s = email.Utils.formataddr((name, addr)).encode(enc, 'replace')
                 print _('Added  : %(s)s')
             except Errors.MMAlreadyAMember:
@@ -276,6 +280,10 @@ def main():
                     # reasons is in the database.  Use a lower level remove to
                     # get rid of this member's entry
                     mlist.removeMember(addr)
+            # Avoid UnicodeError if name can't be decoded
+            if isinstance(name, str):
+                name = unicode(name, errors='replace')
+            name = name.encode(enc, 'replace')
             s = email.Utils.formataddr((name, addr)).encode(enc, 'replace')
             print _('Removed: %(s)s')
 
